legacy/firmware/fsm_msg_ethereum.h

```diff
@@ -282,8 +339,17 @@ void fsm_msgEthereumSignTypedHash(const EthereumSignTypedHash *msg) {
     return;
   }
 
-  if (!fsm_ethereumCheckPath(msg->address_n_count, msg->address_n, false,
-                             CHAIN_ID_UNKNOWN)) {
+  const EthereumDefinitions_encoded_network_t *encoded_network = NULL;
+  if (msg->has_encoded_network) {
+    encoded_network =
+        (const EthereumDefinitions_encoded_network_t *)&msg->encoded_network;
+  }
+
+  const EthereumDefinitionsDecoded *defs = get_EthereumDefinitionsDecoded(
+      encoded_network, NULL, CHAIN_ID_UNKNOWN, NULL);
```

Comment:
...and this one could also be `defs = get_network_only(msg->has_encoded_network, &msg->encoded_network)`, it's repeated three times so it seems worth it.

Reply:
9556b8f18d0dab6fc84e20f31bad8ba707737562